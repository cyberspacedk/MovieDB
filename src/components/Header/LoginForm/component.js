import React from 'react';
import PropTypes from 'prop-types';
import { Form } from 'formik';
import { Button } from 'antd';
import FormField from './FormField/component';

const LoginForm = ({ values, errors, touched }) => (
  <Form>
    <FormField
      type="text"
      name="username"
      placeHolder="Username"
      value={values}
      errors={errors}
      touched={touched}
    />
    <FormField
      type="password"
      name="password"
      placeHolder="Password"
      value={values}
      errors={errors}
      touched={touched}
    />
    <Button
      htmlType="submit"
      type="primary"
      disabled={
        (touched.username && errors.username) ||
        (touched.password && errors.password)
      }
    >
      Login
    </Button>
  </Form>
);

LoginForm.propTypes = {
  values: PropTypes.shape().isRequired,
  errors: PropTypes.shape().isRequired,
  touched: PropTypes.shape().isRequired,
};

export default LoginForm;
