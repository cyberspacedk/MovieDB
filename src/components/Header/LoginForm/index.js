import Loginform from './container';

export default Loginform;
