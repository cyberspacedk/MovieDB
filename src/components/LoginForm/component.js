/* eslint-disable no-unused-vars */
/* eslint-disable no-alert */
/* eslint-disable no-shadow */
/* eslint-disable camelcase */
/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React from 'react';
import axios from 'axios';
import { Form, Icon, Input, Button } from 'antd';
import {
  REQUEST_TOKEN_PATH,
  GET_SESSION_ID_LOGIN_PATH,
  GET_SESSION_ID_PATH,
  DELETE_SESSION_ID_PATH,
} from '../../api/api';
import 'antd/dist/antd.css';

function hasErrors(fieldsError) {
  return Object.keys(fieldsError).some(field => fieldsError[field]);
}

class HorizontalLoginForm extends React.Component {
  componentDidMount() {
    this.props.form.validateFields();
  }

  handleSubmit = e => {
    e.preventDefault();
    this.props.refreshToken(REQUEST_TOKEN_PATH);

    this.props.form.validateFields((err, values) => {
      const { username, password } = values;
      const request_token = JSON.parse(localStorage.getItem('REQUEST_TOKEN'));

      this.props.auth(
        GET_SESSION_ID_LOGIN_PATH,
        username,
        password,
        request_token,
        GET_SESSION_ID_PATH,
      );
    });
  };

  handleLogout = e => {
    // e.preventDefault();
    const SID = JSON.parse(localStorage.getItem('SESSION_ID'));
    this.props.logout(DELETE_SESSION_ID_PATH, SID);
  };

  render() {
    const {
      getFieldDecorator,
      getFieldsError,
      getFieldError,
      isFieldTouched,
    } = this.props.form;

    // Only show error after a field is touched.
    const usernameError =
      isFieldTouched('username') && getFieldError('username');
    const passwordError =
      isFieldTouched('password') && getFieldError('password');

    const { isAuth } = this.props;
    console.log(isAuth);
    return (
      <Form layout="inline" onSubmit={this.handleSubmit}>
        {!isAuth ? (
          <>
            <Form.Item
              validateStatus={usernameError ? 'error' : ''}
              help={usernameError || ''}
            >
              {getFieldDecorator('username', {
                rules: [
                  { required: true, message: 'Please input your username!' },
                ],
              })(
                <Input
                  prefix={
                    <Icon type="user" style={{ color: 'rgba(0,0,0,.25)' }} />
                  }
                  placeholder="Username"
                />,
              )}
            </Form.Item>

            <Form.Item
              validateStatus={passwordError ? 'error' : ''}
              help={passwordError || ''}
            >
              {getFieldDecorator('password', {
                rules: [
                  { required: true, message: 'Please input your Password!' },
                ],
              })(
                <Input
                  prefix={
                    <Icon type="lock" style={{ color: 'rgba(0,0,0,.25)' }} />
                  }
                  type="password"
                  placeholder="Password"
                />,
              )}
            </Form.Item>

            <Form.Item>
              <Button
                type="primary"
                htmlType="submit"
                disabled={hasErrors(getFieldsError())}
              >
                Log in
              </Button>
            </Form.Item>
          </>
        ) : (
          <Form.Item>
            <Button
              type="primary"
              htmlType="submit"
              onClick={this.handleLogout}
            >
              Log out
            </Button>
          </Form.Item>
        )}
      </Form>
    );
  }
}

const WrappedHorizontalLoginForm = Form.create({ name: 'horizontal_login' })(
  HorizontalLoginForm,
);

export default WrappedHorizontalLoginForm;
