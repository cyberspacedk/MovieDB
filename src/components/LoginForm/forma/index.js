import ContainerForm from './container';

export default ContainerForm;
