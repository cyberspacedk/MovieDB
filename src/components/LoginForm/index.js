import FormContainer from './container';

export default FormContainer;
