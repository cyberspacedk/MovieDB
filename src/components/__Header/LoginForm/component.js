import React from 'react';
import { Form, Field } from 'formik';
import FormField from './FormField/component';
import FormButton from './FormButton/component';

const LoginForm = () => (
  <Form>
    <Field name="username" component={FormField} />
    <Field name="password" component={FormField} />
    <Field component={FormButton} />
  </Form>
);

export default LoginForm;
