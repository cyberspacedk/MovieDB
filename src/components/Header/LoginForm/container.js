import { compose } from 'redux';
import { connect } from 'react-redux';
import { withFormik } from 'formik';
import * as Yup from 'yup';
import { authUser } from '../../../store/authentifiction/actions';
import LoginForm from './component';

const mdtp = {
  authUser,
};

export const mapPropsToValues = () => ({
  username: '',
  password: '',
});

// РАЗОБРАТЬСЯ С ФУНКЦИЕЙ САБМИТА
export const handleSubmit = (values, { props, resetForm }) => {
  // eslint-disable-next-line no-shadow
  const { authUser } = props;
  const { rememberMe: _, ...user } = values;
  resetForm();
  authUser(user);
};

export const validationSchema = Yup.object().shape({
  username: Yup.string()
    .min(4, 'Минимум 4 символа')
    .matches(/^[a-z0-9_-]{4,16}$/, 'Запрещенные символы')
    .required('Обязательное поле'),
  password: Yup.string()
    .min(6, 'Минимум 6 символов')
    .required('Обязательное поле'),
});
export default compose(
  connect(
    null,
    mdtp,
  ),
  withFormik({
    mapPropsToValues,
    handleSubmit,
    validationSchema,
  }),
)(LoginForm);
