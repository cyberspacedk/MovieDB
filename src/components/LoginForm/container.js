/* eslint-disable no-shadow */
/* eslint-disable react/prop-types */
import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import {
  sendDataForCheck,
  userLogout,
  newRequestToken,
} from '../../Redux/Actions/setUserDataAction';
import {
  isAuthentificated,
  getUserLogin,
} from '../../Redux/Selectors/authReducer';
import LoginForm from './component';

const Header = styled.header`
  display: flex;
  background-color: #383838;
  padding: 10px 15px;
  margin-bottom: 20px;
  form {
    margin-left: auto;
  }
`;

const ContainerForm = ({
  checkToken,
  logoutUser,
  isAuth,
  refreshToken,
  userLogin,
}) => (
  <Header>
    <LoginForm
      auth={checkToken}
      logout={logoutUser}
      isAuth={isAuth}
      refreshToken={refreshToken}
      userLogin={userLogin}
    />
  </Header>
);
const mstp = state => ({
  isAuth: isAuthentificated(state),
  userLogin: getUserLogin(state),
});
const mdtp = {
  checkToken: sendDataForCheck,
  logoutUser: userLogout,
  refreshToken: newRequestToken,
};

export default connect(
  mstp,
  mdtp,
)(ContainerForm);
