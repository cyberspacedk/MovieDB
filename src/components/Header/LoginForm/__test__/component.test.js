import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import LoginForm from '../component';

describe('Component: LoginForm', () => {
  const props = {
    values: {},
    errors: {
      username: 'some errors in username field',
      password: 'some errors in password field',
    },
    touched: {},
  };

  it('should match its snapshot. Display errors in inputs', () => {
    const LoginFormComponent = shallow(<LoginForm {...props} />);
    expect(shallowToJson(LoginFormComponent)).toMatchSnapshot();
  });

  it('should match its snapshot. Display without errors in inputs', () => {
    const nextProps = {
      ...props,
      errors: {
        username: '',
        password: '',
      },
    };

    const LoginFormComponent = shallow(<LoginForm {...nextProps} />);
    expect(shallowToJson(LoginFormComponent)).toMatchSnapshot();
  });
});
