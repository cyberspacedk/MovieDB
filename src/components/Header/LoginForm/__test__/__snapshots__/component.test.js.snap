// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: LoginForm should match its snapshot. Display errors in inputs 1`] = `
<Form>
  <FormikConnect(FieldInner)
    component={[Function]}
    name="username"
  />
  <FormikConnect(FieldInner)
    component={[Function]}
    name="password"
  />
  <FormikConnect(FieldInner)
    component={[Function]}
  />
</Form>
`;

exports[`Component: LoginForm should match its snapshot. Display errors in inputs 2`] = `
<Form>
  <FormikConnect(FieldInner)
    component={[Function]}
    name="username"
  />
  <FormikConnect(FieldInner)
    component={[Function]}
    name="password"
  />
  <FormikConnect(FieldInner)
    component={[Function]}
  />
</Form>
`;
